aten/src/ATen/native/native_functions.yaml

```diff
@@ -1373,6 +1373,52 @@
 - func: fill_diagonal_(Tensor(a!) self, Scalar fill_value, bool wrap=False) -> Tensor(a!)
   variants: method
 
+- func: diff.Tensor_Tensor(Tensor self, int n=1, int dim=-1, Tensor? prepend=None, Tensor? append=None) -> Tensor
+  variants: function, method
+  dispatch:
+    Math: diff_tensor_tensor
+
+- func: diff.Tensor_Scalar(Tensor self, int n=1, int dim=-1, Tensor? prepend=None, Scalar? append=None) -> Tensor
+  cpp_no_default_args: ['n', 'dim', 'prepend', 'append']
+  variants: function, method
+  dispatch:
+    Math: diff_tensor_scalar
+
+- func: diff.Scalar_Tensor(Tensor self, int n=1, int dim=-1, Scalar? prepend=None, Tensor? append=None) -> Tensor
+  cpp_no_default_args: ['n', 'dim', 'prepend', 'append']
```

Comment:
Can "append" not have a default here? I think prepend being scalar is enough to disambiguate the other overloads.

Reply:
Ah yeah, good catch. But looks like from the comments above, we might end up just landing only the Tensor versions for now.